plugins/football-scoreboard/test_missing_team_logos.py

```
@@ -0,0 +1,129 @@
+    """Pull _ensure_team_logos off sports.py without importing the world."""
+    import ast
+
+    src = (PLUGIN_DIR / "sports.py").read_text(encoding="utf-8")
+    tree = ast.parse(src)
+    fn = next((n for n in ast.walk(tree)
+               if isinstance(n, ast.FunctionDef) and n.name == "_ensure_team_logos"), None)
+    if fn is None:
+        pytest.fail("_ensure_team_logos is gone; missing badges will not be fetched")
+    module = types.ModuleType("_extracted")
+    module.__dict__.update({"Path": Path, "download_missing_logo": None})
+    exec(compile(ast.Module(body=[fn], type_ignores=[]), "sports.py", "exec"), module.__dict__)
```

Ruff (0.16.2)

[error] Use of `exec` detected

(S102)